src/main.c

```diff
@@ -1484,13 +1546,69 @@ static h2o_iovec_t on_extra_status(void *unused, h2o_globalconf_t *_conf, h2o_re
                                           " \"connections\": %d,\n"
                                           " \"max-connections\": %d,\n"
                                           " \"listeners\": %zu,\n"
-                                          " \"worker-threads\": %zu",
+                                          " \"worker-threads\": %zu,\n"
+                                          " \"num-sessions\": %" PRIu64,
                        SSLeay_version(SSLEAY_VERSION), current_time, restart_time, (uint64_t)(now - conf.launch_time), generation,
-                       num_connections(0), conf.max_connections, conf.num_listeners, conf.num_threads);
+                       num_connections(0), conf.max_connections, conf.num_listeners, conf.num_threads, num_sessions(0));
     assert(ret.len < BUFSIZE);
 
+    /* no jemalloc stats? return */
+    if (!JEMALLOC_STATS) {
+        return ret;
+    }
```

Comment:
I _think_ you should create a large `#if JEMALLOC_STATS` surrounding the calls to `malloc_stats_print` and `mallctl` instead of relying on the compiler to strip code paths that never gets executed. Otherwise, linkers might report missing symbol errors when the source file is compiled with optimization turned off.